nengo/cache.py

```diff
@@ -45,6 +45,14 @@ def safe_remove(path):
         logger.warning("OSError during safe_remove: %s", err)
 
 
+def safe_makedirs(path):
+    if not os.path.exists(path):
+        try:
+            os.makedirs(path)
+        except OSError as err:
```

Comment:
Should we make sure that the error is that the file exists here? `if not 'File exists' in str(err): raise` or something like that?